### DISCUSSION TITLE
Inquiry: Storage Reserved Change for Block Type Disks in Engine v2

### DISCUSSION THREAD

**Discussion by jangseon-ryu** [longhorn/longhorn#9838] [Q&A]
Hello Longhorn Community,

I have a question regarding the Storage Reserved setting for block type disks in Longhorn Engine v2. When I attempt to modify the Storage Reserved value for a block type disk, the change is not applied, and an unsupported operation event is triggered.
```
    enginev2-disk1:
      Allow Scheduling:    true
      Disk Driver:         auto
      Disk Type:           block
      Eviction Requested:  false
      Path:                /dev/disk/by-id/scsi-3600508b1001c95d27f513bddea3acc4b
      Storage Reserved:    0  <-- 
      Tags:
```
Log Message:
```
admission webhook "validator.longhorn.io" denied the request: disk enginev2-disk1 type block is not supported to reserve storage
```

Questions:

Does this mean that block type disks do not support changes to the Storage Reserved setting?
If this is the case, are there any plans to support this feature in the future?

Thank you for your assistance. Your guidance would be greatly appreciated.

Best regards,
Jason

**Comment by derekbit** [comment]
What's the purpose of the reservation?

**Reply by jangseon-ryu** [reply]
The reservation serves as a precautionary measure to address the operational limitation that local disks cannot easily accommodate capacity expansion.
So our case is by reserving 20–30% of the total disk capacity, there is a buffer to handle unexpected growth in Persistent Volume Claims (PVCs) or urgent capacity needs.
This conservative approach is part of our operational strategy, rather than a specific feature of Longhorn itself.

**Reply by derekbit** [reply]
WDYT @DamiaSan @shuo-wu @PhanLe1010 @innobead

**Reply by jangseon-ryu** [reply]
Through testing, I have confirmed that Engine v1 (Filesystem type) properly handles changes to Storage Reserved.
If Engine v2 (Block type) also supports this feature but is currently limited to webhooks, I thought it might be possible to support by modifying the webhook configuration.

**Reply by derekbit** [reply]
@jangseon-ryu 
Thanks for the request. Let's convert it to a Feature Request ticket.

**Comment by DamiaSan** [comment]
To implement this feature in v2 engine, I think we should create a virtual disk with the percentage of storage reserved in the physical disk. Because in SPDK, when we create an aio bdev from a disk, the logical volume store we create over it uses the entire disk.
